DataAccess/BookDAO.cs

```diff
@@ -0,0 +1,127 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Group2_BookStore.DB;
+using Group2_BookStore.Models;
+using Microsoft.EntityFrameworkCore;
+
+namespace DataAccess
+{
+    public class BookDAO
+    {
+       private readonly BOOKSTOREContext context; 
+        public BookDAO(BOOKSTOREContext _context) {
+            this.context = _context;
+        }
+        public IEnumerable<Book> GetBookList()
+        {
+            var Books = new List<Book>();
+            try
+            {
+                Books = context.Books.ToList();                
+            }
+            catch (Exception ex)
+            {
+                throw new Exception(ex.Message);
+            }
+            return Books;
+        }
+
+        public IEnumerable<Book> GetBookListByCate(String category)
+        {
+            var Books = new List<Book>();
+            try
+            {
+                Books = context.Books.ToList();                
+            }
+            catch (Exception ex)
+            {
+                throw new Exception(ex.Message);
+            }
+            return Books;
+        }
+        public Book GetBookById(int BookId)
+        {
+            Book Book = null;
+            try
+            {
+                Book = context.Books.SingleOrDefault(c => c.BookId == BookId);
+                if(Book != null){
+                var e = context.Entry(Book);
+                e.Collection(c => c.OrderDetails).Load();
+                e.Reference(p => p.Category).Load();
+                }
+            }
+            catch (Exception ex)
+            {
+                throw new Exception(ex.Message);
+            }
+            return Book;
+        }
+
+        public void AddNew(Book Book)
```

Comment:
```suggestion
        public void AddNew(Book book)
```
Tui nghĩ là nên để tên là `book` để tránh giống model `Book`